CMakeLists.txt

```diff
@@ -863,6 +869,16 @@ if(BUILD_PYTHON_BINDINGS)
   set(ALL_PYTHON_GENERATED_SRCS "") # global list of generated .py files
 
   if(BUILD_PYTHON_PIP_PACKAGE)
+
+    set(PipRepositoryValues "testpypi;pypi" CACHE INTERNAL
+      "List of possible values for which repository to upload to")
+    set(PYTHON_PIP_REPOSITORY "testpypi" CACHE STRING "Which pip repository to use for upload")
+    set_property(CACHE PYTHON_PIP_REPOSITORY PROPERTY STRINGS ${PipRepositoryValues})
+
+    if(NOT PYTHON_PIP_REPOSITORY IN_LIST PipRepositoryValues)
+      message(FATAL_ERROR "PYTHON_PIP_REPOSITORY must be one of ${PipRepositoryValues}")
+    endif()
+
```

Comment:
Another new variable, only if BUILD_PYTHON_PIP_PACKAGE is on to begin with (which is itself only available if BUILD_PYTHON is ON to begin with)